File: app/src/main/java/com/example/chi_12_di/domain/usecase/impl/GetOnePhotoUseCaseImplModule.java
package com.example.chi_12_di.domain.usecase.impl;

import com.example.chi_12_di.domain.usecase.api.IGetOnePhotoUseCase;

import dagger.Module;
import dagger.Provides;

@Module
public class GetOnePhotoUseCaseImplModule {

    @Provides
    IGetOnePhotoUseCase provideGetOnePhotoUseCase (GetOnePhotoUseCaseImpl getOnePhotoUseCaseImpl){
        return getOnePhotoUseCaseImpl;
    }
}

File: app/src/main/java/com/example/chi_12_di/di/GetOnePhotoUseCaseImplModule.kt
package com.example.chi_12_di.di

import com.example.chi_12_di.domain.usecase.api.IGetOnePhotoUseCase
import com.example.chi_12_di.data.usecase.impl.GetOnePhotoUseCaseImpl
import dagger.Binds
import dagger.Module

@Module
interface GetOnePhotoUseCaseImplModule {

    @Binds
    fun  provideGetOnePhotoUseCase (getOnePhotoUseCaseImpl :GetOnePhotoUseCaseImpl ) :IGetOnePhotoUseCase
}
